lib/chat-client/index.js

```diff
@@ -383,7 +383,7 @@ class Chat extends ClientBase {
         } else if (
           // fire onMessage if it was from a different sender or at least a different device
           // from this sender. Bots can filter out their own messages from other devices.
-          (!channel || channel.name === messageObject.msg.channel.name) &&
+          (!channel || channel.topicName === messageObject.msg.channel.topicName) &&
```

Comment:
I think we still need to check `name` here, and other fields in `channel`. Perhaps have a helper function [like this](https://github.com/keybase/kirabot/blob/fa96e8d323db85cc674bd1803b115ae50a715769/src/kira.js#L50-L54)?

Reply:
good catch on the name.  It would need to be a bit more complicated than your helper function, because it would need to understand the defaults. For example, the `public` field and some of the other fields are optional, and users may not pass these in. 

But yeah, I'll add this, thanks for catching.

Reply:
Oh good point! WDYT about exporting that helper function so library users can use?

Also, not suggesting such a change in this PR but for future I think this would be solved by `immutable.js` where we populate default values when creating objects just like in FE code.

Reply:
`immutable.js` brings some other benefit though. For example, when comparing channels we can do a simple reference comparison with `===` and if it's the same we know it's the same channel, and if not we'd compare the fields.

But yeah, it'd be a big opinionated dependency and if we do that we'd probably want to do it for other types too which would be a big change. Not sure; just thinking out loud!

Reply:
@pzduniak agreed about spawning separate processes. Although, on the other hand, I dislike how the bot needs to process everything that comes through keybase chat. If you just want to watch one channel, it stinks your bot needs to get everything from every team and ever channel and filter it out. I almost wish the chat api listen call took a channel as an argument. (But that would keep us on this multi-process model.)

Song, I think I dislike bringing immutable in because it's, as you say, massive, and we're trying to keep this lightweight and simple. `===` won't work anyway. These channel comparisons don't actually match. A user asks to listen to {name:"keybase",topicType:"general"} and what comes back is {name:"keybase",public:false,topicType:"general",membersType:"team", // etc. So somewhere in this code we need to understand the idea of defaults and map their abbreviated request to a full options object. If we do that, then we can use `===` but then we're just going through all the steps of comparing anyway. 

anyway, lame of me to try to race this out the door and break the team check. I'll get this fixed today before merging.